js/ui/form/ui.form.item_options_actions.js

```diff
@@ -48,6 +64,19 @@ const tryCreateItemOptionAction = (optionName, itemActionOptions) => {
             return new ValidationRulesItemOptionAction(itemActionOptions);
         case "cssClass":
             return new CssClassItemOptionAction(itemActionOptions);
+        case "badge":
+        case "disabled":
+        case "icon":
+        case "template":
+        case "tabTemplate":
+        case "title": {
+            const { item, itemsRunTimeInfo } = itemActionOptions;
+            const widgetInstance = itemsRunTimeInfo.findWidgetInstanceByItem(item);
+            if(widgetInstance && widgetInstance.NAME === "dxTabPanel") {
```

Comment:
This is a long round trip for an external data, call the findWidgetInstanceByItem method of an object from  the itemsRunTimeInfo property of the itemActionOptions argument.

The [findWidgetInstanceByItem](https://github.com/DevExpress/DevExtreme/blob/30a545d0e7ae11d51a34410c38e5e42c4e286c42/js/ui/form/ui.form.items_runtime_info.js#L73) implementation is this:
    findWidgetInstanceByItem(item) {
        return this._findWidgetInstance(storedItem => storedItem === item);
    }

The [_findWidgetInstance](https://github.com/DevExpress/DevExtreme/blob/30a545d0e7ae11d51a34410c38e5e42c4e286c42/js/ui/form/ui.form.items_runtime_info.js#L9) implementation is this:
    _findWidgetInstance(condition) {
        let result;

        each(this._map, function(guid, { widgetInstance, item }) {
            if(condition(item)) {
                result = widgetInstance;

                return false;
            }
        });

        return result;
    }

The _map is filled via addItemsOrExtendFrom/add...
And, what is an object from the 'item' property of the itemActionOptions argument? Why it is appropriate to pass it to the findWidgetInstanceByItem method? I am lost in all these 'calls that calls' in a very short time.

It seems to be safer to place calculations out of this method and pass a result as argument.